### ISSUE TITLE
Xcode 14 beta fails when build a Realm v10.28.1 watchOS Target 

### ISSUE THREAD

**Issue by rubencalde** [realm/realm-swift#7848]
### How frequently does the bug occur?

All the time

### Description

Hello, 

When trying to build the Realm binary of a **watchOS target**, the _build.sh_ script of the 10.28.1 version, returns the error described below:
`ld: symbol(s) not found for architecture arm64`

Is this error related to the Issue:
> When you archive a watchOS app with the [arm64e](https://developer.apple.com/documentation/security/preparing_your_app_to_work_with_pointer_authentication) architecture enabled, it fails to build. (93550623)

which is one of the Known Issues on [Xcode 14 beta Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes
)?

This behaviour it seems related by gongzhang post too, on https://github.com/realm/realm-swift/pull/7829#issuecomment-1152845908

Is this commit, https://github.com/realm/realm-swift/pull/7829, confirming that?:
![disable watchos](https://user-images.githubusercontent.com/57009570/173812803-deaa687b-e233-49e4-a022-572933ec651e.png)

Has anyone successfully built the Realm watchOS target with Xcode 14 beta?
Please, we need this problem to be solved as soon as possible. Not being able to compile the watchOS target with Xcode 14 is being very blocking for our work team. Thanks in advance.

Regards

### Stacktrace & log output

```shell
...
NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for realm::TableView", referenced from:
      realm::TableView::TableView(realm::TableView&&) in RLMAccessor.o
      realm::TableView::~TableView() in RLMAccessor.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "non-virtual thunk to realm::ArrayIntNull::get_any(unsigned long) const", referenced from:
      vtable for realm::BPlusTree<realm::util::Optional<long long> >::LeafNode in RLMQueryUtil.o
  "non-virtual thunk to realm::ArrayInteger::get_any(unsigned long) const", referenced from:
      vtable for realm::BPlusTree<long long>::LeafNode in RLMQueryUtil.o
  "non-virtual thunk to realm::ArrayFixedBytes<realm::UUID, 16>::calc_byte_len(unsigned long, unsigned long) const", referenced from:
      vtable for realm::BPlusTree<realm::util::Optional<realm::UUID> >::LeafNode in RLMQueryUtil.o
      vtable for realm::BPlusTree<realm::UUID>::LeafNode in RLMQueryUtil.o
  "non-virtual thunk to realm::ArrayFixedBytes<realm::ObjectId, 12>::calc_byte_len(unsigned long, unsigned long) const", referenced from:
      vtable for realm::BPlusTree<realm::util::Optional<realm::ObjectId> >::LeafNode in RLMQueryUtil.o
      vtable for realm::BPlusTree<realm::ObjectId>::LeafNode in RLMQueryUtil.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **
The following build commands failed:
	Ld /Users/user/Desktop/realm-swift-10.28.1/build/DerivedData/Realm/Build/Intermediates.noindex/Realm.build/Release-watchos/Realm.build/Objects-normal/arm64/Binary/Realm normal arm64 (in target 'Realm' from project 'Realm')
(1 failure)
```

### Can you reproduce the bug?

Yes, always

### Reproduction Steps

Execute the script
`build.sh xcframework watchos`
on a Realm 10.28.1 version project

### Version

10.28.1

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

watchOS 9

### Build environment

Xcode version: 14 beta 
Dependency manager and version: Cocoapods 1.10.2
macOS version: Monterey - 12.4

**Comment by rubencalde** [comment]
Hi, 

I'm getting the same error on Xcode 14 beta 2.
Is there any plan to fix this?

Regards

Rubén

**Build environment**
Xcode version: 14 beta 2
Dependency manager and version: Cocoapods 1.10.2
macOS version: Monterey - 12.4

**Comment by rubencalde** [comment]
More info: 

In addition to the problems described above, although it's possible to correctly build the watchOS target on Xcode 14, when validating a podspec of a project that has a dependency on RealmSwift, the following error is returned:

` error: Cannot code sign because the target does not have an Info.plist file and one is not being generated automatically. Apply an Info.plist file to the target using the INFOPLIST_FILE build setting or generate one automatically by setting the GENERATE_INFOPLIST_FILE build setting to YES (recommended). (in target 'App' from project 'App') `

Similarly, you also get the same error when validating the RealmSwift.podspec, with the following command:

`pod spec lint ./RealmSwift.podspec --allow-warnings sources=master --verbose`

It seems the same error described on https://github.com/CocoaPods/CocoaPods/issues/11402

Build environment
Xcode version: 14 beta 2
Dependency manager and version: Cocoapods 1.10.2
macOS version: Monterey - 12.4

**Comment by tgoyne** [comment]
https://github.com/realm/realm-core/pull/5577 adds a arm64 slice for watchOS to the prebuilt library. The workaround for using the current release is to remove the arm64 architecture from the supported architectures for your app.

**Comment by rubencalde** [comment]
After getting the same error mentioned above, with RealmSwift 10.28.1:
`error: Cannot code sign because the target does not have an Info.plist file and one is not being generated automatically. Apply an Info.plist file to the target using the INFOPLIST_FILE build setting or generate one automatically by setting the GENERATE_INFOPLIST_FILE build setting to YES (recommended). (in target 'App' from project 'App')
    ** BUILD FAILED **`
We were able to fix this by updating to the latest version of CocoaPods, 1.11.3 with version 1.22.0 of the xcodeproj gem. In this way it has been possible to validate the RealmSwift podspec

Rubén

**Build environment**
Xcode version: 14 beta 2
Dependency manager and version: Cocoapods 1.11.3 + xcodeproj 1.22.0
macOS version: Monterey - 12.4

**Comment by jordimele** [comment]
Hello,

I have a similar problem, because when updating an existing application, which uses a watchOS target, to Xcode14 beta2 with RealmSwift 10.28.2, I cannot compile the application.

<img width="618" alt="Build_error" src="https://user-images.githubusercontent.com/11957195/177280964-0a7d0313-3c27-4393-8d6f-feb4bfed36f9.png">

@tgoyne Is it possible that version RealmSwift 10.28.2 does not have support for watchOS or tvOS?
[possible issue](https://github.com/realm/realm-swift/issues/7842)

Are you working on solving the problem? Is there any workaround?

Thanks.

Build environment
Xcode 14 beta 2
Cocoapods 1.11.3
RealmSwift 10.28.2
macOS Monterey 12.4

**Comment by dianaafanador3** [comment]
Hi @jordimele as mentioned on Xcode 14 Beta 2 [release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes) this is no longer supported.
`* Building iOS projects with deployment targets for the armv7, armv7s, and i386 architectures is no longer supported. (92831716)`

**Comment by dianaafanador3** [comment]
Closing this, as this is working in our latest RealmSwift release [v10.28.2](https://github.com/realm/realm-swift/releases/tag/v10.28.2), which updates core to v12.3.0 which fix this issue.

**Comment by mr-v** [comment]
@dianaafanador3 I'm still seeing the "ld: symbol(s) not found for architecture arm64" error both with v10.28.2 and latest `master`. This happens when I'm building for a device.

To reproduce:

A) 
Build the watchOS target from installation examples (`examples/installation/watchos/swift/CocoaPodsExample`) with Xcode 14 beta 3:

<img width="445" alt="image" src="https://user-images.githubusercontent.com/830743/177929396-6f59d6b6-53b2-44f6-9806-38322dcbce12.png">

B)
- set the Xcode 14 beta 3 path as the active developer directory with `xcode-select`,
- run ` ./build.sh xcframework watchos` script.

**Comment by tgoyne** [comment]
This is fixed in core master, which has not made it into a release yet.

**Comment by dianaafanador3** [comment]
@mr-v fix Closing this as the fix was released in `v10.28.4`
